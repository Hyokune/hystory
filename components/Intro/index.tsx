import Image from 'next/image'
import Link from 'next/link'
import React from 'react'

const Intro = () => (
  <div className="intro">
    <h1 className="center">Hello, I am Lance</h1>
    <p className="center">I'm currently a front-end developer aspiring to become a game developer</p>
    <div className="social-about">
      <p>Follow me</p>
      <ul className="social-links">
        <li>
          <a href="https://github.com/Hyokune" target="_blank" rel="noopener noreferrer">
            <i className="fab fa-github" />
          </a>
        </li>
        <li>
          <a href="https://twitter.com/hyokune" target="_blank" rel="noopener noreferrer">
            <i className="fab fa-twitter" />
          </a>
        </li>
        <li>
          <a href="https://www.linkedin.com/in/hyuck-lee-8104a0ba" target="_blank" rel="noopener noreferrer">
            <i className="fab fa-linkedin-in" />
          </a>
        </li>
        <li>
          <Link href="/resume">
            <a>
              <i className="far fa-id-badge" />
            </a>
          </Link>
        </li>
        <li>
          <a href="mailto:lleehyuck@gmail.com">
            <i className="far fa-envelope" />
          </a>
        </li>
      </ul>
    </div>
    <Image unsized src="/images/lance-alternative.png" alt="Lance alternative drawing by Kitoodles" />
  </div>
)

export default Intro
