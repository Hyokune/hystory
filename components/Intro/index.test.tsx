import { shallow } from 'enzyme'
import React from 'react'
import Intro from './index'

describe('Intro component', () => {
  test('render component without error', () => {
    const wrapper = shallow(<Intro />)
    expect(wrapper).toMatchSnapshot()
    expect(wrapper.hasClass('intro')).toBe(true)
  })
})
