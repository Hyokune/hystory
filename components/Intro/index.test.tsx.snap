// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Intro component render component without error 1`] = `
<div
  className="intro"
>
  <h1
    className="center"
  >
    Hello, I am Lance
  </h1>
  <p
    className="center"
  >
    I'm currently a front-end developer aspiring to become a game developer
  </p>
  <div
    className="social-about"
  >
    <p>
      Follow me
    </p>
    <ul
      className="social-links"
    >
      <li>
        <a
          href="https://github.com/Hyokune"
          rel="noopener noreferrer"
          target="_blank"
        >
          <i
            className="fab fa-github"
          />
        </a>
      </li>
      <li>
        <a
          href="https://twitter.com/hyokune"
          rel="noopener noreferrer"
          target="_blank"
        >
          <i
            className="fab fa-twitter"
          />
        </a>
      </li>
      <li>
        <a
          href="https://www.linkedin.com/in/hyuck-lee-8104a0ba"
          rel="noopener noreferrer"
          target="_blank"
        >
          <i
            className="fab fa-linkedin-in"
          />
        </a>
      </li>
      <li>
        <Link
          href="/resume"
        >
          <a>
            <i
              className="far fa-id-badge"
            />
          </a>
        </Link>
      </li>
    </ul>
  </div>
  <Image
    alt="Lance alternative drawing by Kitoodles"
    src="/images/lance-alternative.png"
    unsized={true}
  />
</div>
`;
